telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/InlineKeyboardMarkup.kt

```diff
@@ -1,16 +1,36 @@
 package com.github.kotlintelegrambot.entities
 
 import com.github.kotlintelegrambot.entities.keyboard.InlineKeyboardButton
-import com.google.gson.Gson
+import com.github.kotlintelegrambot.network.serialization.GsonFactory
 import com.google.gson.annotations.SerializedName
 
-data class InlineKeyboardMarkup(
+/**
+ * This object represents an inline keyboard that appears right next to the message it belongs to.
+ * @param inlineKeyboard Array of button rows, each represented by an Array of [InlineKeyboardButton] objects.
+ * @see <https://core.telegram.org/bots/api#inlinekeyboardmarkup>
+ */
+class InlineKeyboardMarkup internal constructor(
     @SerializedName("inline_keyboard") val inlineKeyboard: List<List<InlineKeyboardButton>>
 ) : ReplyMarkup {
+
+    init {
+        // Buttons of type [Pay] must always be the first button in the first row.
+        val flattenedButtons = inlineKeyboard.flatten()
+        val payButtons = flattenedButtons.filterIsInstance<InlineKeyboardButton.Pay>()
+        require(payButtons.size <= 1) { "Can't have more than one pay button per inline keyboard" }
+        require(payButtons.size != 1 || flattenedButtons.firstOrNull() is InlineKeyboardButton.Pay) {
```

Comment:
hmmm... I see your point but IMO there are some reasons why we shouldn't do that: 
* It would be a bit weird to alter the ordering of the keyboard keys automagically and could be annoying for the clients of this API.
* We'll be letting users to use the API incorrectly without noticing it (because we'll be fixing the errors under the hood). 
* It wouldn't be consistent because we'll be just fixing the incorrect position of the PAY button but not fixing having more than one pay button.

Also, I think this "offensive" approach for validation is more widely used and most of the library users would expect it in this way.